spec/models/dinosaur_spec.rb

```diff
@@ -1,5 +1,181 @@
+        expect(dino.errors[:enjoys]).to include "is too short (minimum is 10 characters)" 
+    end
+
+    it 'is not valid if image is less than 10 characters long' do
+        dino = Dinosaur.create(
+        name: 'Triceratops',    
+        age: 69000000,
+        diet: 'Herbavore(Vegan)',
+        enjoys: 'Enjoys long walks in the rainforest and using its large skull to make a path.',
+        image: 'https://')
+
+        expect(dino.errors[:image]).to_not be_empty
+        expect(dino.errors[:image]).to include "is too short (minimum is 10 characters)" 
+    end
+
+    it 'does not allow duplicate names' do
+        Dinosaur.create(
+        name: 'Triceratops',    
+        age: 69000000,
+        diet: 'Herbavore(Vegan)',
+        enjoys: 'Enjoys long walks in the rainforest and using its large skull to make a path.',
+        image: 'https://static.wikia.nocookie.net/jurassicpark/images/5/52/Jurassic_world_fallen_kingdom_triceratops_by_sonichedgehog2-dc9dwcu.png/revision/latest?cb=20180427200649')
+
+        dino = Dinosaur.create(
+        name: 'Triceratops',    
+        age: 69000000,
+        diet: 'Herbavore(Vegan)',
+        enjoys: 'Enjoys long walks in the rainforest and using its large skull to make a path.',
+        image: 'https://static.wikia.nocookie.net/jurassicpark/images/5/52/Jurassic_world_fallen_kingdom_triceratops_by_sonichedgehog2-dc9dwcu.png/revision/latest?cb=20180427200649')
+
+        expect(dino.errors[:name]).to be_empty
+        expect(dino.errors[:name]).to_not include "has already been taken" 
+    end
+
+    it 'does not allow duplicate image' do
+        Dinosaur.create(
+        name: 'Triceratops',    
+        age: 69000000,
+        diet: 'Herbavore(Vegan)',
+        enjoys: 'Enjoys long walks in the rainforest and using its large skull to make a path.',
+        image: 'https://static.wikia.nocookie.net/jurassicpark/images/5/52/Jurassic_world_fallen_kingdom_triceratops_by_sonichedgehog2-dc9dwcu.png/revision/latest?cb=20180427200649')
+
+        dino = Dinosaur.create(
+        name: 'Triceratops',    
+        age: 69000000,
+        diet: 'Herbavore(Vegan)',
+        enjoys: 'Enjoys long walks in the rainforest and using its large skull to make a path.',
+        image: 'https://static.wikia.nocookie.net/jurassicpark/images/5/52/Jurassic_world_fallen_kingdom_triceratops_by_sonichedgehog2-dc9dwcu.png/revision/latest?cb=20180427200649')
+
+        expect(dino.errors[:image]).to be_empty
+        expect(dino.errors[:image]).to_not include "has already been taken" 
+    end
+
 end
+
+
+
+
+# {
+#     name: 'Triceratops',
+#     age: 69000000,
+#     diet: 'Herbavore(Vegan)',
+#     enjoys: 'Enjoys long walks in the rainforest and using its large skull to make a path.',
+#     image: 'https://static.wikia.nocookie.net/jurassicpark/images/5/52/Jurassic_world_fallen_kingdom_triceratops_by_sonichedgehog2-dc9dwcu.png/revision/latest?cb=20180427200649'
+# },
```

Comment:
Please remove unused commented out code.

Reply:
Commented out code removed. Thank you